currencies/rate_converter_test.go

```diff
@@ -13,354 +13,273 @@ import (
+			time.Sleep(200 * time.Millisecond)
+
+			assert.Equal(t, (time.Time{}), currencyConverter.LastUpdated(), formatTestErrorMsg(test.description, "LastUpdated() shouldn't be set"))
+			assert.Equal(t, &currencies.ConstantRates{}, currencyConverter.Rates(), formatTestErrorMsg(test.description, "Conversions.Rates weren't the expected ones"))
+			continue
+		}
 
-	// Setup:
-	calledURLs := []string{}
-	mockedHttpServer := httptest.NewServer(http.HandlerFunc(
-		func(rw http.ResponseWriter, req *http.Request) {
-			calledURLs = append(calledURLs, req.RequestURI)
-			rw.WriteHeader(http.StatusOK)
-			rw.Write(nil)
-		}),
-	)
+		lastUpdated := currencyConverter.LastUpdated()
+		assert.NotEqual(t, (time.Time{}), lastUpdated, formatTestErrorMsg(test.description, "LastUpdated() should return a time set"))
+		assert.True(t, lastUpdated.After(beforeExecution), formatTestErrorMsg(test.description, "LastUpdated() should be after last update"))
 
-	defer mockedHttpServer.Close()
+		rates := &currencies.Rates{}
+		err := json.Unmarshal(test.mockResponse, rates)
+		assert.NoError(t, err, formatTestErrorMsg(test.description, "JSON unmarshalling of conversions failed"))
 
-	// Execute:
-	currencyConverter := currencies.NewRateConverter(
-		&http.Client{},
-		mockedHttpServer.URL,
-		time.Duration(24)*time.Hour,
-	)
+		assert.Equal(t, rates, currencyConverter.Rates(), formatTestErrorMsg(test.description, "Rates() doesn't return expected rates"))
 
-	// Verify:
-	assert.Equal(t, 1, len(calledURLs), "sync URL should have been called %d times but was %d", 1, len(calledURLs))
-	assert.Equal(t, currencyConverter.LastUpdated(), (time.Time{}), "LastUpdated() shouldn't return a time set")
-	assert.Nil(t, currencyConverter.Rates(), "Rates() should return nil")
+		// Check for the next 1 second that no fetch was triggered
+		time.Sleep(500 * time.Millisecond)
+		assert.Equal(t, lastUpdated, currencyConverter.LastUpdated(), formatTestErrorMsg(test.description, "Currency rates were fetched before the requested duration was passed"))
+	}
 }
 
-func TestInit(t *testing.T) {
-
+func TestNewRateConverterWithNotifier(t *testing.T) {
 	// Setup:
-	mockedHttpServer := httptest.NewServer(http.HandlerFunc(
-		func(rw http.ResponseWriter, req *http.Request) {
-			rw.WriteHeader(http.StatusOK)
-			rw.Write([]byte(
-				`{
-					"dataAsOf":"2018-09-12",
-					"conversions":{
-						"USD":{
-							"GBP":0.77208
-						},
-						"GBP":{
-							"USD":1.2952
-						}
-					}
-				}`,
-			))
-		}),
-	)
-
-	// Execute:
-	expectedTicks := 5
-	ticksTimes := []time.Time{}
-	ticks := make(chan int)
-	currencyConverter := currencies.NewRateConverterWithNotifier(
-		&http.Client{},
-		mockedHttpServer.URL,
-		time.Duration(100)*time.Millisecond,
-		ticks,
-	)
+	rates := &currencies.Rates{}
+	err := json.Unmarshal(getMockRates(), rates)
+	assert.NoError(t, err, "JSON unmarshalling of conversions failed")
+
+	mockServerHandler := func(mockResponse []byte, code int, initialFailCount int,
```

Comment:
**Style**: The type for `code` and for `initialFailCount` does not need to be specified since they're the same as `failRespCode`.